lib/features/email/presentation/controller/single_email_controller.dart

```diff
@@ -1708,6 +1715,19 @@ class SingleEmailController extends BaseController with AppLoaderMixin {
     }
   }
 
+  void _maybeCalendarEventAction() {
+    if (_maybeCalendarEventInteractor == null) return;
+    if (_displayingEventBlobId == null) return;
+
+    if (mailboxDashBoardController.accountId.value != null) {
+      consumeState(_maybeCalendarEventInteractor!.execute(
+        mailboxDashBoardController.accountId.value!,
+        {_displayingEventBlobId!}));
+    } else {
+      consumeState(Stream.value(Left(CalendarEventMaybeFailure())));
+    }
```

Comment:
IMO, should return errors in if conditions.

```suggestion
if (_maybeCalendarEventInteractor == null 
|| _displayingEventBlobId == null 
|| mailboxDashBoardController.accountId.value == null) {
   consumeState(Stream.value(Left(CalendarEventMaybeFailure())));
   return;
} else {
consumeState(_maybeCalendarEventInteractor!.execute(
    mailboxDashBoardController.accountId.value!,
    {_displayingEventBlobId!})
);
}
```

Reply:
Done